File: receiver.cpp
#include <opencv/build/include/opencv2/opencv.hpp>
#include <iostream>
using namespace std;
using namespace cv;


int receiver(){
    //TODO 1. 인풋 스트림을 열어서
    // stream = Gobject.~~~
    // frame = stream.read()
    //TODO 2. 수신 받은 걸 openCV에 쓸 수 있도록 가공

    //TODO 3. openCV에서 돌려서 결과출력
    Mat frame;

    while(true) {

        cap.read(frame);

        if(frame.empty())
            break;

        imshow("Receiver", frame);
        if(waitKey(1) == 'r')
            break;
    }
    destroyWindow("Receiver");
}

File: receiver.py
import cv2 as cv
import numpy as np

cap = cv.VideoCapture('udpsrc port=9777 ! application/x-rtp ! rtph264depay ! h264parse ! avdec_h264 ! videoconvert ! appsink', cv.CAP_GSTREAMER)

if not cap.isOpened():
    print("VideoCapture not opened")
    exit(-1)

while True:
    ret, frame = cap.read()
    if not ret:
        print("empty frame")
        break
    cv.imshow("Receiver", frame)
    if cv.waitKey(1) == 'r':
        break

cap.release()
cv.destroyWindow("Receiver")